### PR TITLE
Fix #1264: Avoid unnecessary art resizes

### LINKED ISSUE #1264: Re-importing with fetchart creates two album art files
Heya guys,

Not a bug or anything but I'm occasionally getting multiple cover art files.
I'm thinking it's when I previously had the incorrect musicbrainz ID for the album, thus the existing `cover.jpg` isn't valid.

an example: http://pastebin.com/mz13gBZ9

I'm `cover.2.jpg` is the correct art for the release of that album I have. 
I'm kind of anal here and would much rather have any existing `cover*.jpg` files deleted if it's going to get new art.

I already have `*.jpg` configured under `clutter:`, perhaps incorrectly thinking it would remove any jpgs when it checked for new artwork.

_I suspect I'm just not groking something in the beets decision tree_

fyi: fetchart is enabled and runs with these settings:

```
fetchart:                                   # get album art
    auto:               yes                 # default: yes
    cautious:           yes                 # require cover_names
    cover_names:        cover folder front picard  # be paranoid
    sources:            coverart itunes *   # prefer coverart itunes
    maxwidth:           2500                # max of 2500px wide please
    remote_priority:    no                  # prefer local
```

### PR DESCRIPTION
**Pull request by sampsyo**
Part of #1679 by @reiv.

> I believe this is caused by the fact that if a local `cover.jpg` exists, the `fetchart` plugin would then "discover" this during the import and reuse it. Furthermore, if the `maxwidth` setting is used, `fetchart` will always resize it, creating `cover.1.jpg` in the process. I've rewritten it slightly so that it will only resize if the image is actually wider than `maxwidth`, making `import -L` an idempotent operation.

### CODE REVIEW

File: beetsplug/fetchart.py [hunk 1]
```diff
@@ -41,6 +41,10 @@
 CONTENT_TYPES = ('image/jpeg', 'image/png')
 DOWNLOAD_EXTENSION = '.jpg'
 
+CANDIDATE_BAD = 0
+CANDIDATE_EXACT = 1
+CANDIDATE_DOWNSCALE = 2
```
Comment (sampsyo) [inline beetsplug/fetchart.py]: We do now have real enums via the `enum34` backport module—perhaps we should use that.

File: beetsplug/fetchart.py [hunk 2]
```diff
@@ -506,10 +510,10 @@ def _fetch_image(self, url):
 
     def _is_valid_image_candidate(self, candidate):
```
Comment (sampsyo) [inline beetsplug/fetchart.py]: This method needs a docstring describing the three possible return values.

**Comment by sampsyo** [comment]
This is also a good idea independent of file duplication—we shouldn't be resizing images that are small enough already. :+1:

A test here should be pretty straightforward; we could use a mock to verify that the resize function doesn't get called.

**Comment by reiv** [comment]
Made some fixes and added a test case mocking the resize function as you suggested.

File: beetsplug/fetchart.py (hunk 1 above)
Comment (reiv) [inline beetsplug/fetchart.py]: enums sound a little overkill for something that I presume will only be used internally by the plugin.

Comment (sampsyo) [inline beetsplug/fetchart.py]: Yeah, point taken.

**Comment by sampsyo** [comment]
Looks perfect; thanks again! :sparkles:

**Pull request merged**